src/services/trackers/mangaupdate.ts

```diff
@@ -0,0 +1,793 @@
+      .then((data: TokenResponseData) => {
+        if (data.status === 'exception') {
+          log.error(
+            `Error getting token from tracker ${MUTrackerMetadata.id}: 
+              ${data.reason}`
+          );
+          return null;
+        }
+        return data.context!.session_token;
+      })
+      .catch((e: Error) => {
+        log.error(e);
+        return null;
+      });
+  };
+
+  getUsername: GetUsernameFunc = () => {
+    if (this.accessToken === '') return new Promise((resolve) => resolve(null));
+
+    const url = `${BASE_URL}/account/profile`;
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${this.accessToken}`,
+        Accept: 'application/json',
+      },
+    };
+
+    return fetch(url, options)
+      .then((response) => {
+        if (response.status === 401) {
+          log.error(
+            `Error getting username from tracker ${MUTrackerMetadata.id}: Unauthorized access`
+          );
+          return null;
+        }
+        return response.json();
+      })
+      .then((data: UserResponseData | null) => {
+        return data ? data.username : null;
+      })
+      .catch((e: Error) => {
+        log.error(e);
+        return null;
+      });
+  };
+
+  search: SearchFunc = async (text: string) => {
+    if (this.accessToken === '') return new Promise((resolve) => resolve([]));
+
+    const url = `${BASE_URL}/series/search`;
+    const options = {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${this.accessToken}`,
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      body: JSON.stringify({ search: text }),
+    };
+
+    return fetch(url, options)
+      .then((response) => response.json())
+      .then((data: MangaListResponseData) => {
+        return data.results.map((item) => {
+          const { record } = item;
+          return {
+            id: record.series_id.toString(),
+            title: record.title,
+            description: record.description,
+            coverUrl: record.image.url.original,
+          } as TrackerSeries;
+        });
+      })
+      .catch((e: Error) => {
+        log.error(e);
+        return [];
+      });
+  };
+
+  getLibraryEntry: GetLibraryEntryFunc = async (seriesId: string) => {
+    if (this.accessToken === '') return new Promise((resolve) => resolve(null));
+
+    const url = `${BASE_URL}/lists/series/${seriesId}`;
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${this.accessToken}`,
+        Accept: 'application/json',
+      },
+    };
+
+    return fetch(url, options)
+      .then((response) => {
+        if (response.status === 404) {
+          log.error(
+            `Error getting library entry for series ${seriesId} from tracker ${MUTrackerMetadata.id}: Series not found`
+          );
+          return null;
+        }
+        else if (response.status === 401) {
```

Comment:
else is unnecessary when the above block always returns. A few other instances in this file can be removed too